src/platform/plugins/shared/vis_types/timeseries/public/application/visualizations/views/metric.js

```diff
@@ -11,12 +11,125 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import _ from 'lodash';
 import { EuiResizeObserver } from '@elastic/eui';
+import { css } from '@emotion/react';
 
 import { getLastValue } from '../../../../common/last_value_utils';
 import { calculateCoordinates } from '../lib/calculate_coordinates';
 import { RenderCounter } from '../../components/render_counter';
 
-import './_metric.scss';
+import { getVisVariables } from './_variables';
+
+/**
+ * 1. Text is scaled using a matrix so all font sizes and related metrics
+ *    are being calculated from a percentage of the base font size of 100% (14px).
+ */
+
+// Helper function to create styles for reversed elements
+const createReversedStyles = (styles) => css`
+  .tvbVisMetric--reversed & {
+    ${styles}
+  }
+`;
+
+const containerStyle = css`
+  font-size: 100%; /* 1 */
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  flex: 1 0 auto;
+`;
+
+const resizeStyle = css`
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  overflow: hidden;
+  display: flex;
+`;
+
+const innerStyle = ({ euiTheme }) => css`
+  position: absolute;
+  padding: ${euiTheme.size.xs} ${euiTheme.size.s} ${euiTheme.size.s};
+`;
+
+export const labelPrimaryStyle = ({ euiTheme }) => {
+  const vars = getVisVariables({ euiTheme });
+
+  return css`
+    color: ${vars.tvbTextColor};
+    text-align: center;
+    font-size: 0.5em; /* 1 */
+    margin-bottom: 0.25em; /* 1 */
+    line-height: 1em; /* 1 */
+
+    ${createReversedStyles(`color: ${vars.tvbTextColorReversed};`)}
+  `;
+};
+
+const valuePrimaryStyle = ({ euiTheme }) => {
+  const vars = getVisVariables({ euiTheme });
+
+  return css`
+    color: ${vars.tvbValueColor};
+    text-align: center;
+    font-size: 1em; /* 1 */
+    font-weight: ${euiTheme.font.weight.bold};
+    line-height: 1em; /* 1 */
+
+    ${createReversedStyles(`color: ${vars.tvbValueColorReversed};`)}
+  `;
+};
+
+export const secondaryContainerStyle = css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-top: 0.05em; /* 1 */
+`;
+
+export const labelSecondaryStyle = ({ euiTheme }) => {
+  const vars = getVisVariables({ euiTheme });
+
+  return css`
+    font-size: 0.35em; /* 1 */
+    margin-right: 0.3em; /* 1 */
+    color: ${vars.tvbTextColor};
+    line-height: 1em; /* 1 */
+
+    ${createReversedStyles(`color: ${vars.tvbTextColorReversed};`)}
+  `;
+};
+
+export const valueSecondaryStyle = ({ euiTheme }) => {
+  const vars = getVisVariables({ euiTheme });
+
+  return css`
+    font-size: 0.35em; /* 1 */
+    color: ${vars.tvbValueColor};
+    line-height: 1em; /* 1 */
+
+    ${createReversedStyles(`color: ${vars.tvbValueColorReversed};`)}
+  `;
+};
+
+const labelAdditionalStyle = ({ euiTheme }) => {
+  const vars = getVisVariables({ euiTheme });
+
+  return css`
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```

Comment:
Are you doing this instead of `euiTextTruncate()` because of the `createReversedStyles` thing?